File: tdd/Counter.m
#import "Counter.h"

NSString *const CounterModelChanged = @"CounterModelChanged";

@implementation Counter

- (void)increment {
    _count += 1;
    [self postNotification];
}

- (void)decrement {
    _count -= 1;
    [self postNotification];
}

- (void)postNotification {
    [[NSNotificationCenter defaultCenter] postNotificationName:CounterModelChanged
                                                        object:self];
}
@end

File: quick-test-swift/quick-test-swift/Counter.swift
import Foundation

public let CounterModelChanged = "CounterModelChanged"

public class Counter {
    public var count:Int?

    public init() {
        count = 0
    }

    public func increment() {
        count!++
        NSNotificationCenter.defaultCenter().postNotificationName(CounterModelChanged, object:self)
    }

    public func decrement() {
        count!--
        NSNotificationCenter.defaultCenter().postNotificationName(CounterModelChanged, object:self)
    }
}
